### ISSUE TITLE
`zarf package create . -f typo` => misleading error message

### ISSUE THREAD

**Issue by JoeHCQ1** [zarf-dev/zarf#2659]
### Environment
Device and OS: Sys76 Laptop (PopOS)
App/package versions: Zarf 0.35.0
Kubernetes distro being used: k3d
Other:

### Steps to reproduce
1. Clone a working UDS package such as the [mattermost package](https://github.com/defenseunicorns/uds-package-mattermost).
2. From the root directory, attempt to package it with a typo in your `-f` value: `zarf package create . -f stream` (should be `upstream`)

### Expected result

The error message tells me that the value after the `-f` flag is invalid.

### Actual Result

The error message tells me the problem is in my zarf.yaml file.

### Visual Proof (screenshots, videos, text, etc)

![image](https://github.com/defenseunicorns/zarf/assets/49208786/1d0a08ea-ee4c-4236-9ded-7bce1836503a)

### Severity/Priority

Low, but I think it's not benign as an unclear error may lead someone to stop experimenting with the application and so hurt adoption.

### Additional Context
Add any other context or screenshots about the technical debt here.

**Comment by AustinAbro321** [comment]
I agree with this as I've also been confused in a similar way in the past. Likely [Compose Components](https://github.com/defenseunicorns/zarf/blob/main/src/pkg/packager/creator/compose.go#L16) should return an error if there are no compatible components.

**Comment by AustinAbro321** [comment]
language has been changed to `failed to create package: package validation failed: package does not contain any compatible components`
Additionally, this is somewhat of a duplicate of #2587. If we provided input validation for a flavor that does not exist in the package it would be immediately clear what the issue is from a user's perspective
